build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -1690,4 +1690,22 @@ message RequestMetadata {
   // An identifier to tie multiple tool invocations together. For example,
   // runs of foo_test, bar_test and baz_test on a post-submit of a given patch.
   string correlated_invocations_id = 4;
+
+  // A one-word description of the kind of action, for example, CppCompile or GoLink.
+  // There is no standard agreed set of values for this, and they are expected to vary between different client tools.
+  string action_mnemonic = 5;
+
+  // A namespace within which the build was performed, for instance a repository/workspace name.
+  // There are no guarantees that these are globally unique, e.g. two users may create namespaces with the same name.
+  string invocation_namespace_id = 6;
```

Comment:
If it's per-invocation, putting it in the per-action metadata is *not* ideal. I've previously seen cases where metadata ended up being more than 50% of the network traffic, with most of it being redundant. Please don't.

Reply:
That's a reasonable concern - the same is presumably true for `correlated_invocations_id` - I expect that's tied to a single `tool_invocation_id`?

Perhaps both of these fields could be attached to `GetCapabilities` (either as in-line metadata in the request, or as header metadata) so that a remote execution system can record an association between them, and then they could be omitted from the per-request headers?

Reply:
An alternative would be to only attach the bare minimum here, and rely on the Build Event Stream to provide all other info. More concretely, this info should be part of the metadata:
- invocation ID
- target name
- configuration ID
But this info could be part of the BES:
- correlated invocations ID
- tool name and version

Reply:
If you happen to be running a BES, I think that makes complete sense, but I'm not sure closely coupling those two systems is broadly applicable enough to have one depend on the other.

Reply:
Can we leave this field out for now and check the rest in? :-D The other new items seem obviously a good idea to add to the proto.

Reply:
Totally - updated to remove that field :)